src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -1099,7 +1106,37 @@ static int xgq_collect_sensor_data(struct xocl_xgq *xgq)
 		kfree(cmd);
 	}
 
-	return ret;
+	return xgq->sensor_data;
```

Comment:
If xgq_collect failed, the sensor data will be invalid.

Reply:
it will have all 0's, so not a problem. In client hwmon_sdm driver, it check for 1st byte (completion_code), it reads 0, means SDR_CODE_NOT_AVAILABLE. It just return, don't read any other index in buffer.